histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -157,51 +226,57 @@ def ComputeHaralickFeatures(im_label, im_intensity, offsets=None, num_levels=8,
 
     # List of feature names
     feature_list = [
-        'Haralick.ASM.Mean',
-        'Haralick.Contrast.Mean',
-        'Haralick.Correlation.Mean',
-        'Haralick.SumofSquar.Mean',
-        'Haralick.IDM.Mean',
-        'Haralick.SumAverage.Mean',
-        'Haralick.SumVariance.Mean',
-        'Haralick.SumEntropy.Mean',
-        'Haralick.Entropy.Mean',
-        'Haralick.Variance.Mean',
-        'Haralick.DifferenceEntropy.Mean',
-        'Haralick.IMC1.Mean',
-        'Haralick.IMC2.Mean',
-        'Haralick.ASM.Range',
-        'Haralick.Contrast.Range',
-        'Haralick.Correlation.Range',
-        'Haralick.SumofSquar.Range',
-        'Haralick.IDM.Range',
-        'Haralick.SumAverage.Range',
-        'Haralick.SumVariance.Range',
-        'Haralick.SumEntropy.Range',
-        'Haralick.Entropy.Range',
-        'Haralick.Variance.Range',
-        'Haralick.DifferenceEntropy.Range',
-        'Haralick.IMC1.Range',
-        'Haralick.IMC2.Range',
+        'Haralick.ASM',
+        'Haralick.Contrast',
+        'Haralick.Correlation',
+        'Haralick.SumOfSquare',
+        'Haralick.IDM',
+        'Haralick.SumAverage',
+        'Haralick.SumVariance',
+        'Haralick.SumEntropy',
+        'Haralick.Entropy',
+        'Haralick.DifferenceVariance',
+        'Haralick.DifferenceEntropy',
+        'Haralick.IMC1',
+        'Haralick.IMC2',
     ]
 
-    # compute object properties if not provided
-    if rprops is None:
-        rprops = regionprops(im_label)
+    agg_feature_list = []
+    for fname in feature_list:
+        agg_feature_list.append(fname + '.Mean')
+        agg_feature_list.append(fname + '.Range')
 
-    # create pandas data frame containing the features for each object
-    numFeatures = len(feature_list)
-    numLabels = len(rprops)
-    fdata = pd.DataFrame(np.zeros((numLabels, numFeatures)),
-                         columns=feature_list)
+    # num_levels
+    if num_levels is None:
+        num_levels = _default_num_levels(im_intensity)
 
     # check for consistent shapes between 'I' and 'Label'
     if im_intensity.shape != im_label.shape:
         raise ValueError("Inputs 'I' and 'Label' must have same shape")
 
-    # determine if image is grayscale or RGB
-    if len(im_intensity.shape) != 2:  # color image
-        raise ValueError("Inputs 'I' should be a grayscale image")
```

Comment:
This function will work on ND single channel images. User of this function is supposed to make sure that he passes in a single-channel image.